src/main/java/de/internetsicherheit/brl/bloxberg/cache/ethereum/BlockWithData.java

```diff
@@ -46,4 +50,14 @@ public BlockWithData(EthBlock.Block ethBlock) {
         this.timestamp = ethBlock.getTimestamp();
     }
 
+    public List<BlockTransaction> getTransactions() {
```

Comment:
This is one approach, but this is indeed potentially expensive for the user of the method (I would not expect a getter method to be this involved).

IMO doing this work already in the constructor and then having `List<BlockTransaction>` as the private field is easier and better. Or it is at least how I as and outside user of the API would expect the code to behave.

Reply:
So, the constructor should have a the private field eg List<BlockTransaction>  block transactions? But how can someone from the outside of the constructor have access to it to make use of that list?

Reply:
The constructor converts data from `EthBlock.Block` into our internal data structures. So the body of your current `getTransactions()` method would basically move into the constructor body (or a private method called by the constructor).

We can keep the getter method `getTransactions()`, however it should simply return the internal field `private List<BlockTransaction> transactions`, which has already been populated in the constructor.